allennlp_models/generation/models/copynet_seq2seq.py

```diff
@@ -142,9 +140,19 @@ def __init__(
         self._output_copying_layer = Linear(self.encoder_output_dim, self.decoder_output_dim)
 
         # At prediction time, we'll use a beam search to find the best target sequence.
-        self._beam_search = BeamSearch(
-            self._end_index, max_steps=max_decoding_steps, beam_size=beam_size
+        # For backwards compatibility, check if beam_size or max_decoding_steps were passed in as
+        # kwargs. If so, update the BeamSearch object before constructing and raise a DeprecationWarning
+        deprecation_warning = (
+            "The parameter {} as been deprecated."
+            "Provide this parameter as argument to beam_search instead."
         )
+        if "beam_size" in kwargs:
+            beam_search._params["beam_size"] = kwargs["beam_size"]
+            raise DeprecationWarning(deprecation_warning.format("beam_size"))
+        if "max_decoding_steps" in kwargs:
+            beam_search._params["max_steps"] = kwargs["max_decoding_steps"]
+            raise DeprecationWarning(deprecation_warning.format("max_decoding_steps"))
+        self._beam_search = beam_search.construct(end_index=self._end_index)
```

Comment:
@epwalsh Here is my solution for addressing your last outstanding comment about backwards compatibility (will do it for the rest of the models if you are happy with this).

Reply:
Hmm I'd rather not modify `beam_search._params` since it's a private attribute. Instead I would just make a `beam_search_extras = {}` variable and add `"beam_size"` and `"max_decoding_steps"` from `kwargs` into it, if found. Then do `beam_search.construct(end_index=self._end_index, **beam_search_extras)`.

Reply:
Right, good catch. Done! Shall I update the other generation models with this change?

Reply:
Looks good! Go ahead with that and then we can get this merged ✅

Reply:
Awesome. Done. I had to tweak some of the tests under `generation`. Looks like AllenNLP treats `DeprecationWarning` as an error? From the `pytest.ini`:

"Our policy here is to ignore (silence) any deprecation warnings from _outside_ allennlp, but to treat any _internal_ deprecation warnings as errors."

That caused some trouble when a model config had `beam_size` or `max_decoding_steps`, as my fix catches those and then calls `DeprecationWarning`, causing the test to fail. In any case, all tests under `generation/models` and `generation/modules` are passing locally now.

Reply:
Oh, interesting. I'm not sure when we decided that, but I'm going to disable that for now.

Reply:
Ah, nevermind, I just updated those test fixtures